File: Práctica4/FigurasGeometricas.py
import math

class FiguraGeometrica:
    def area(self, *args):
        if len(args) == 1:
            radio = args[0]
            return math.pi * radio ** 2  # Círculo
        elif len(args) == 2:
            base, altura = args
            return base * altura  # Rectángulo o paralelogramo
        elif len(args) == 3:
            base, altura, _ = args
            return (base * altura) / 2  # Triángulo
        elif len(args) == 4:
            base1, base2, altura, _ = args
            return ((base1 + base2) * altura) / 2  # Trapecio
        elif len(args) == 3 and isinstance(args[2], bool):
            perimetro, apotema, _ = args
            return (perimetro * apotema) / 2  # Hexágono
        else:
            raise ValueError("Número de argumentos no válido")

# Creación de instancias y pruebas
f = FiguraGeometrica()

print("Círculo:", f.area(1))
print("Rectángulo:", f.area(2, 3))
print("---------------------TAREA----------------")
print("Triángulo Rectángulo:", f.area(3, 5, 0))
print("Trapecio:", f.area(4, 6, 7, True))
print("Hexágono:", f.area(3, 5, True))

File: Práctica4/FigurasGeometricas.java
package figurageométrica;

public class FiguraGeométrica {

    double area (double radio){
        return Math.PI * radio * radio;
    } 
    
    double area (double base, double altura)
    {
        return base * altura;
    }
    
    double area(double base, double altura, double triangulo)
    {
        return (base * altura) / 2;
    }

    double area(double base1, double base2, double altura, boolean trapecio)  
    {
        return ((base1 + base2) * altura) / 2;
    }

    double area(double perimetro, double apotema, boolean hexagono)
    {
        return (perimetro * apotema) / 2;
    }

    
    public static void main(String[] args) {
        // TODO code application logic here
        FiguraGeométrica f1 = new FiguraGeométrica();
        FiguraGeométrica f2 = new FiguraGeométrica();
        
        FiguraGeométrica f3 = new FiguraGeométrica();
        FiguraGeométrica f4 = new FiguraGeométrica();
        FiguraGeométrica f5 = new FiguraGeométrica();
        
        System.out.println("Circulo: "+f1.area(1));
        System.out.println("Rectangulo: "+f2.area(2,3));
        
        System.out.println("---------------------TAREA----------------");
        System.out.println("Triangulo Rectangulo: "+f3.area(3,5));
        System.out.println("Trapecio: "+f4.area(4,6,7));
        System.out.println("Hexagono: "+f5.area(3,5));
        
    }
}